src/services/FileSystemService.ts

```diff
@@ -27,15 +27,28 @@ export class FileSystemService implements IProjectFilesBrowserService {
     return fs.promises.writeFile(path, content);
   }
 
-  createDirectory(path: string) {
+  async createDirectory(path: string) {
+  if (!(await this.isDirectory(nodePath.dirname(path)))) {
```

Comment:
Isn't it always true? I thought the createDirectory() receives the dirname path as the parameter so it doesn't exist yet. Or how does it work?